.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,91 @@
+# This workflow will run the integration tests for the project
+name: Tests - Integration
+
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        description: 'The name of the artifact to download'
+        required: true
+        type: string
+    secrets:
+      copilot-key:
+        description: 'The Copilot key to use for integration tests'
+        required: true
+
+jobs:
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.12" ]
+    steps:
+      - name: Download Docker image artifact
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4
+        with:
+          name: ${{ inputs.artifact-name }}
+
+      - name: Load Docker image
+        run: |
+          docker load -i image.tar
+          echo "Loaded image:"
+          docker images
+
+      - name: Run container from the loaded image
+        run: |
+          # Get the image name
+          IMAGE_NAME=$(docker images --format "{{.Repository}}:{{.Tag}}" | head -n 1)
+          echo "Running container from image: $IMAGE_ID"
+
+          # Run the container
+          docker run --name codegate -d -p 8989:8989 -p 9090:9090 \
+            -p 8990:8990 \
+            --restart unless-stopped ghcr.io/stacklok/codegate:latest
+
+          # Confirm the container started
+          echo "Container started:"
+          docker ps
+
+          # Verify container is running with correct ports
+          docker ps -f name=codegate
+
+          # Check mount configuration
+          docker inspect codegate -f '{{ json .Mounts }}' | jq
+
+          # Verify volume contents are accessible
+          docker exec codegate ls -la /app/codegate_volume
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          lfs: true
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Poetry
+        uses: snok/install-poetry@76e04a911780d5b312d89783f7b1cd627778900a # v1
+        with:
+          version: 2.0.1
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('**/poetry.lock') }}
+
+      - name: Install dependencies
+        run: poetry install --with dev
+
+      - name: Run integration tests
+        env:
+          CODEGATE_PROVIDERS: "copilot"
+          CA_CERT_FILE: "/home/runner/work/codegate/codegate/codegate_volume/certs/ca.crt"
```

Comment:
You might be able to fix the ca stuff with something like this:

```bash
cp ca.crt /usr/local/share/ca-certificates/codegate.crt
update-ca-certificates
```

I don't know if sudo would be needed, do these run as root in github?

Reply:
We should be able to either run the tests without installing the CA_CERT_FILE or not specify the CA_CERT_FILE argument if we had installed the cert to the system store.